x/authority/types/authorizations_test.go

```diff
@@ -0,0 +1,196 @@
+package types_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	"github.com/zeta-chain/zetacore/x/authority/types"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	lightclienttypes "github.com/zeta-chain/zetacore/x/lightclient/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func TestAuthorizationList_SetAuthorizations(t *testing.T) {
```

Comment:
Would be better to change to Table-Driven tests. This would also simplify in the review checking the assertions are correct
{
  CurrentTable,
  NewValue,
  ExpectedTable
}

Reply:
I had originally considered a table , but discarded it as it looked more messy in my opinions specially considering , that we have about 2 test cases per Test 

Here is a sample of what that function would look like, I do prefer to keep this style of tests as in my opinion it looks easier to read , but happy to change if it makes for an easier review 

```
func TestAuthorizationList_SetAuthorizations(t *testing.T) {
	tt := []struct {
		name             string
		oldList          types.AuthorizationList
		addAuthorization types.Authorization
		expectedList     types.AuthorizationList
	}{
		{
			name:             "set new authorization successfully",
			oldList:          types.DefaultAuthorizationsList(),
			addAuthorization: sample.Authorization(),
			expectedList: types.AuthorizationList{
				Authorizations: append(types.DefaultAuthorizationsList().Authorizations, sample.Authorization()),
			},
		},
		{
			name:    "update existing authorization successfully",
			oldList: types.AuthorizationList{Authorizations: []types.Authorization{sample.Authorization()}},
			addAuthorization: types.Authorization{
				MsgUrl:           sample.Authorization().MsgUrl,
				AuthorizedPolicy: types.PolicyType_groupEmergency,
			},
			expectedList: types.AuthorizationList{
				Authorizations: []types.Authorization{
					{
						MsgUrl:           sample.Authorization().MsgUrl,
						AuthorizedPolicy: types.PolicyType_groupEmergency,
					},
				},
			},
		},
	}

	for _, tc := range tt {
		t.Run(tc.name, func(t *testing.T) {
			tc.oldList.SetAuthorizations(tc.addAuthorization)
			require.Equal(t, tc.expectedList, tc.oldList)
		})
	}
}
```